modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/filter/expression/PredicateExpressionVisitorImpl.java

```diff
@@ -630,11 +630,18 @@ private Object _getValue(
 		EntityField.Type entityType = entityField.getType();
 
 		if (entityType.equals(EntityField.Type.DATE_TIME) &&
-			(DBManagerUtil.getDBType() == DBType.HYPERSONIC)) {
+			(Objects.equals(DBManagerUtil.getDBType(), DBType.HYPERSONIC) ||
+			 Objects.equals(DBManagerUtil.getDBType(), DBType.ORACLE))) {
+
+			String pattern = "dd-MMM-yyyy HH:mm:ss.SSS";
+
+			if (Objects.equals(DBManagerUtil.getDBType(), DBType.ORACLE)) {
+				pattern = StringUtil.replace(pattern, "HH", "hh") + " a";
```

Comment:
@p-albuquerque I would change to use the string instead, like: `pattern = "dd-MMM-yyyy hh:mm:ss.SSS a";`